src/Cli/dotnet/commands/dotnet-package/search/NugetSearchApiRequest.cs

```diff
@@ -0,0 +1,174 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using LocalizableStrings = Microsoft.DotNet.Tools.Package.Search.LocalizableStrings;
+
+namespace Microsoft.DotNet.Cli.commands.package.search
+{
+    internal class NugetSearchApiRequest
+    {
+        private readonly string _sourceSeparator = new string('=', 20);
+        private readonly string _packageSeparator = new string('-', 20);
+        private string _searchTerm;
+        private int? _skip;
+        private int? _take;
+        private bool _prerelease;
+        private List<string> _sources;
+        private bool _exactMatch;
+
+        public NugetSearchApiRequest(string searchTerm, int? skip, int? take, bool prerelease, bool exactMatch, List<string> sources)
+        {
+            _searchTerm = searchTerm;
+            _skip = skip;
+            _take = take;
+            _prerelease = prerelease;
+            _sources = sources;
+            _exactMatch = exactMatch;
+        }
+        public async void ExecuteCommandAsync()
+        {
+            var taskList = new List<(Task<IEnumerable<IPackageSearchMetadata>>, PackageSource)>();
+            
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            WarnForHTTPSources(listEndpoints);
+            foreach (PackageSource source in listEndpoints)
+            {
+                Console.WriteLine($"Name {source.Source}");
+                SourceRepository repository = Repository.Factory.GetCoreV3(source);
+                Console.WriteLine($"Name {repository.PackageSource}");
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>(CancellationToken.None);
+                Console.WriteLine($"Name {source.Name}");
+                if (resource is null)
+                {
+                    Console.WriteLine($"{source.Name} 2");
+                    taskList.Add((null, source));
+                    continue;
+                }
+                Console.WriteLine($"{source.Name}3");
+                taskList.Add((Task.Run(() => resource.SearchAsync(
+                    _searchTerm,
+                    new SearchFilter(includePrerelease: _prerelease),
+                    skip: _skip ?? 0,
+                    take: _take ?? 20,
+                    NullLogger.Instance,
+                    CancellationToken.None
+                    )), source));
+                Console.WriteLine($"{source.Name}4");
+            }
+            Console.WriteLine("wow " + taskList.Count().ToString() );
+            foreach (var taskItem in taskList)
+            {
+                Console.WriteLine("WTF");
```

Comment:
You should squash it before merging, so it's not part of the commit history for the repo :)